msal_extensions/token_cache.py

```diff
@@ -0,0 +1,196 @@
+"""Generic functions and types for working with a TokenCache that is not platform specific."""
+import os
+import sys
+import time
+import errno
+import msal
+from .cache_lock import CrossPlatLock
+
+if sys.platform.startswith('win'):
+    from .windows import WindowsDataProtectionAgent
+elif sys.platform.startswith('darwin'):
+    from .osx import Keychain
+
+
+def get_protected_token_cache(
+        cache_location=None,
+        lock_location=None,
+        enforce_encryption=False, **kwargs):
+    """Detects the current system, and constructs a TokenCache of the appropriate type for the
+    environment in which it's running.
+
+    :param cache_location: The name of the file holding the serialized TokenCache.
+    :param lock_location: The file that should be used to ensure different TokenCache using
+    processes are not racing one another.
+    :param enforce_encryption: When 'True' an error will be raised if there isn't an encrypted
+    option available for the current system. When 'False', a plain-text option will be returned.
+    :param kwargs: Any options that should be passed to the platform-specific constructor of the
+    TokenCache being instantiated by this method.
+    :return: A fully instantiated TokenCache able to encrypt/decrypt tokens on the current system.
+    """
+    if sys.platform.startswith('win'):
+        return WindowsTokenCache(cache_location, lock_location, **kwargs)
+
+    if sys.platform.startswith('darwin'):
+        return OSXTokenCache(cache_location, lock_location, **kwargs)
+
+    if enforce_encryption:
+        raise RuntimeError('no protected token cache for platform {}'.format(sys.platform))
+
+    return FileTokenCache(cache_location, cache_location, **kwargs)
+
+
+def _mkdir_p(path):
+    """Creates a directory, and any necessary parents.
+
+    This implementation based on a Stack Overflow question that can be found here:
+    https://stackoverflow.com/questions/600268/mkdir-p-functionality-in-python
+
+    If the path provided is an existing file, this function raises an exception.
+    :param path: The directory name that should be created.
+    """
+    try:
+        os.mkdir(path)
```

Comment:
I believe you mean to use [`os.makedirs()`](https://docs.python.org/2/library/os.html#os.makedirs)?

Reply:
Nice catch! Fixed in d3059c37617d75d1c583615fc65ad95e3036a99e